### ISSUE TITLE
user-callback script is not executed when running cron job

### ISSUE THREAD

**Issue by klayman** [bit-team/backintime#723]
I have created a test `user-callback` file in the `~/.config/backintime/` directory and it is executed with correct arguments when running `backintime-qt4` (opening and closing the GUI, taking snapshot etc.).

But when the scheduled backup cron job is running, the `user-callback` script is not used.

I have no idea of how to debug this problem. I've just modified the crontab to save the stdout to a text file, and the first two lines in that file were:
```
"sni-qt/15901" WARN  22:01:42.271 void StatusNotifierItemFactory::connectToSnw() Invalid interface to SNW_SERVICE 
WARNING: Inhibit Suspend failed.
```
could these warnings relate to the problem? By the way, the tray icon when running backup job from GUI differs from the icon when a cron job is running - may be this could help to find the cause of the problem.

**Comment by klayman** [comment]
Ok, I solved the problem.

I was trying to mount a drive with `udisksctl` command in the `user-callback` script. To use it in a cron job there should be an appropriate polkit rule for it (if  polkit is active). I created the file `/etc/polkit-1/localauthority/50-local.d/10-my-udisks-rules.pkla:
```
[Local Users]
Identity=unix-user:*
Action=org.freedesktop.udisks*
ResultAny=yes
```
and the drive was successfully mounted.

Additionally, I was testing the `user-callback` script with `notify-send` command. To use it in a cron job, there should be the following inside the `user-callback`:
```
eval "export $(egrep -z DBUS_SESSION_BUS_ADDRESS /proc/$(pgrep -u $LOGNAME openbox)/environ)";
export DISPLAY=:0
```
where `openbox`  should be replaced with the active session (e.g. `gnome-session`).
